Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -466,27 +466,47 @@ internal async Task<BoolResult> RestoreCheckpointAsync(OperationContext context,
 
         private async Task<BoolResult> ProcessStateAsync(OperationContext context, bool inline)
         {
-            try
+            var result = await processStateCoreAsync();
+
+            if (result.Succeeded &&

```

Comment:
Technically this is a race, since multiple ProcessStateAsyncs could be running. Clear that's not a problem 99.99% of the time, but if it does ever happen that two are finishing at the same time, is that a problem?

Reply:
We have only one operation running at the same time.
And even if they can, I don't think we can ended up with an inconsistent state.

Field assignment is atomic and task transition is thread safe. One thread definitely can see that the task is not null, but after that it can't get into a null state. So we can't get NRE.

The worst thing that may happen here is double assignment when two operations are done at the same time and two threads will decide to update the field.

---
In reply to: [323499999](https://github.com/microsoft/BuildXL/pull/862#discussion_r323499999) [](ancestors = 323499999)